haystack/fields.py

```diff
@@ -473,6 +473,58 @@ def convert(self, value):
 
         return [value]
 
+    def resolve_attributes_lookup(self, current_objects, attributes):
+        """
+        Return the leaf node values for one or more iterable objects that contain data,
+        for model attributes that can be multiple objects (relations) deep.
+        """
+        values = []
+
+        for current_object in current_objects:
+            while len(attributes) > 1:
+                if not hasattr(current_object, attributes[0]):
+                    raise SearchFieldError(
+                        "The model '%s' does not have a model_attr '%s'."
+                        % (repr(current_object), attributes[0])
+                    )
+                current_object = getattr(current_object, attributes.pop(0))
+
+                if hasattr(current_object, "all") and ismethod(current_object.all):
+                    attribute = "__".join(attributes)
+                    return list(
+                        current_object.values_list(attribute, flat=True).distinct()
+                    )
+
+                elif not hasattr(current_object, "__iter__"):
+                    if hasattr(current_object, attributes[0]):
```

Comment:
Are there tests that come with these changes?  What will happen here if attributes is empty?

Reply:
@cclauss That code will run only when `len(attributes) > 1`, so I don't think there would be an issue?

Reply:
You are correct.  My mistake.